doc/tutorial/basic/tutorial.rst

```diff
@@ -227,10 +228,9 @@ Unless otherwise specified, input will be cast to ``float64``::
   >>> transformer = kernel_approximation.RBFSampler()
   >>> X_new = transformer.fit_transform(X)
   >>> X_new.dtype
-  dtype('float64')
+  dtype('float32')
 
-In this example, ``X`` is ``float32``, which is cast to ``float64`` by
-``fit_transform(X)``.
+In this example, ``X`` is ``float32``, and is unchanged by ``fit_transform(X)``.
```

Comment:
```suggestion
In this example, ``X`` is ``float32``, and is unchanged by ``fit_transform(X)``.

Using `float32`-typed training (or testing) data is often more 
efficient than using the usual ``float64`` ``dtype``: it allows to
reduce the memory usage and sometimes also reduces processing time
by leveraging the vector instructions of the CPU. However it can
sometimes lead to numerical stability problems causing the algorithm
to be more sensitive to the scale of the values and :ref:`require
adequate preprocessing<preprocessing_scaler>`.

Keep in mind however that not all scikit-learn estimators attempt to
work in `float32` mode. For instance, some transformers will always
cast there input to `float64` and return `float64` transformed
values as a result.
```